### PR TITLE
Added ignorePatterns and includePatterns options

### PR DESCRIPTION
**Pull request by papigers**
Solves #40 
- Added `ignorePatterns` option to ignore certain SVG files from processing (blacklist).
- Added `includePatterns` option to process only certain files (whitelist).

These options cannot be used together.

### CODE REVIEW

**Comment by aeirola** [comment]
Thanks for the PR! In general looks good, but would maybe be good to discuss the configuration format a bit.

The react-native default tooling, such as jest (https://jestjs.io/docs/en/configuration#testregex-string--arraystring), flow (https://flow.org/en/docs/config/ignore/) and metro (https://facebook.github.io/metro/docs/configuration/#blacklistre) all seem to prefer using RegExps over other formats such as the `.gitignore` pattern. So while I also personally do like the gitignore pattern format, I'm a bit cautious about deviating from the convention set by the react native tooling.

What kind thoughts does this provoke for a fellow react native developer?

**Comment by papigers** [comment]
For me personally glob patterns are more readable and easier to understand. But I see where you're coming from, I'll try to update the PR tomorrow to use regex instead.

**Comment by papigers** [comment]
Hope that's better :)

File: src/index.js [hunk 1]
```diff
@@ -81,6 +118,10 @@ async function convertSvg(assetData: AssetData): Promise<AssetData> {
   const inputFilePath = assetData.files[0];
   const inputFileScale = assetData.scales[0];
 
+  if (ignores(inputFilePath)) {
```
[the remark this answers is not in the document]
Comment (papigers) [reply src/index.js:121]: Good point, will do

File: README.md [hunk 2]
```diff
@@ -78,6 +78,8 @@ Where the possible configuration values are:
 | `cacheDir` | string   | `'.png-cache'` | Name of directory to store cached PNGs. |
 | `scales`   | number[] | `[1, 2, 3]`    | PNG image scales to generate for different screen densities. |
 | `output`   | object   | `{}`           | Sharp PNG output [options](http://sharp.pixelplumbing.com/en/v0.22.1/api-output/#png). |
+| `ignoreRegex`   | RegExp   | `null`           | Regex that will be matched against the source file's full path, if there's a match the file will be ignored. (Cannot be used together with `includeRegex`).  |
+| `includeRegex`   | RegExp   | `null`           | Regex that will be matched against the source file's full path, the file will be processed only if there's a match. (Cannot be used together with `ignoreRegex`). |
```
[the remark this answers is not in the document]
Comment (papigers) [reply README.md:82]: Yea I was migrating the glob pattern configuration into this, not realizing that regex is powerful enough to make these separate configurations unnecessary.

**Comment by aeirola** [comment]
Thanks again. I'll do some chores on the repo, and the release this promptly.

**Comment by aeirola** [comment]
This is now released in version 0.4.0. Thanks again for the help!

**Comment by papigers** [comment]
Awesome! Thank you!

**Pull request merged**